--- netcdf-c-4.8.0/nczarr_test/ncdumpchunks.c.old
+++ netcdf-c-4.8.0/nczarr_test/ncdumpchunks.c
@@ -10,7 +10,7 @@
 #include <getopt.h>
 #endif
 
-#ifdef _WIN32
+#if !defined(__GETOPT_H__)
 #include "XGetopt.h"
 #endif
 
--- netcdf-c-4.8.0/nczarr_test/s3util.c.old
+++ netcdf-c-4.8.0/nczarr_test/s3util.c
@@ -13,7 +13,7 @@
 #ifdef HAVE_GETOPT_H
 #include <getopt.h>
 #endif
-#ifdef _WIN32
+#if !defined(__GETOPT_H__)
 #include "XGetopt.h"
 #endif
 
--- netcdf-c-4.8.0/nczarr_test/tst_utils.c.old
+++ netcdf-c-4.8.0/nczarr_test/tst_utils.c
@@ -14,7 +14,7 @@
 #include <getopt.h>
 #endif
 
-#ifdef _WIN32
+#if !defined(__GETOPT_H__)
 #include "XGetopt.h"
 #endif
 
--- netcdf-c-4.8.0/nczarr_test/zmapio.c.old
+++ netcdf-c-4.8.0/nczarr_test/zmapio.c
@@ -13,7 +13,7 @@
 #ifdef HAVE_GETOPT_H
 #include <getopt.h>
 #endif
-#ifdef _WIN32
+#if !defined(__GETOPT_H__)
 #include "XGetopt.h"
 #endif
 
--- netcdf-c-4.8.0/nczarr_test/zs3parse.c.old
+++ netcdf-c-4.8.0/nczarr_test/zs3parse.c
@@ -11,7 +11,7 @@
 #ifdef HAVE_GETOPT_H
 #include <getopt.h>
 #endif
-#ifdef _WIN32
+#if !defined(__GETOPT_H__)
 #include "XGetopt.h"
 #endif
 
